pywbem/_subscription_manager.py

```diff
@@ -1253,29 +1276,31 @@ def _create_filter(self, server_id, source_namespace, query,
         # Validate server_id
         server = self._get_server(server_id)
 
-        this_host = getfqdn()
         ownership = "owned" if owned else "permanent"
 
-        filter_path = CIMInstanceName(FILTER_CLASSNAME,
-                                      namespace=server.interop_ns)
+        svr_name = self._get_systemname(server_id)
 
         filter_inst = CIMInstance(FILTER_CLASSNAME)
-        filter_inst.path = filter_path
         filter_inst['CreationClassName'] = FILTER_CLASSNAME
         filter_inst['SystemCreationClassName'] = SYSTEM_CREATION_CLASSNAME
-        filter_inst['SystemName'] = this_host
+        # Issue #2704, dummy set because mock requires the property
+        filter_inst['SystemName'] = svr_name or "dummy"
```

Comment:
If `_get_systemname()` is changed as suggested, its return value will never be `None`, and the `or "dummy"` can be removed.

Reply:
DONE